Framework/PythonInterface/mantid/kernel/CMakeLists.txt

```diff
@@ -131,6 +131,10 @@ add_library(PythonKernelModule
             ${PYTHON_INSTALL_FILES})
 set_python_properties(PythonKernelModule _kernel)
 
+# GCC 8 onwards needs to disable functional casting at the Python interface
+target_compile_options( PythonKernelModule PRIVATE 
```

Comment:
As far as I can tell this affects the places where a function pointer with a default parameter is cast to a function pointer with no parameters. I think this only affects the `Material.cpp` file? Should we suppress it more locally with the macros in `WarningSuppressions.h` instead? Or the other way around this in `Material.cpp` would be to put in a local function that takes no arguments and delegates to `Material::cohScatterXSection` and the others. That way we wouldn't need to suppress the warning?

Am I missing something?

Reply:
@DavidFair Have you had a chance to look at this.

Reply:
Not yet, planning on resolving this today or tomorrow. Thanks for the nudge as I had forgotten about it